src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DesignerSupport.targets

```diff
@@ -128,12 +128,18 @@ Copyright (c) .NET Foundation. All rights reserved.
       <_DesignerShadowCopy Include="@(ReferenceCopyLocalPaths)" />
 
       <!--
-        For .NET Core, we do not include NuGet package assets or runtime pack
-        assets, irrespective of self-contained / copy-local settings. Designer
+        For .NET Core, we do not include NuGet package assets,
+        irrespective of self-contained / copy-local settings. Designer
         will load these from the NuGet cache or shared framework always.
       -->
       <_DesignerShadowCopy
-        Remove="@(_ResolvedCopyLocalBuildAssets);@(RuntimePackAsset)"
+        Remove="@(_ResolvedCopyLocalBuildAssets)"
```

Comment:
At this point the `_DesignerShadowCopy` list includes all of the runtime pack assets and the files from NuGet packages, and we want to remove those (except the runtime packs in the case of RuntimePackAlwaysCopyLocal).  I think what you want is something like this (assuming I got the syntax right):

```xml
      <!-- For .NET Core, we do not include NuGet package assets, as the designer will load these from the NuGet cache. -->
      <_DesignerShadowCopy
        Remove="@(_ResolvedCopyLocalBuildAssets)"
        Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'"
        />

      <!-- For .NET Core runtime pack assets, we do not include them regardless of whether the app is
              self-contained, as they will be loaded from the shared framework.  However, for runtime pack
              assets where RuntimePackAlwaysCopyLocal is true, there is no shared framework, so we still
              include those.  -->
      <_DesignerShadowCopy
        Remove="@(RuntimePackAsset)"
        Condition="'%(RuntimePackAsset.RuntimePackAlwaysCopyLocal)' != 'true'"
        />

```

Reply:
Thanks, I have updated this file as per your suggestion and confirmed that its working fine. I have also updated the test to run for `SelfContained` apps.
I remember testing `_DesignerShadowCopy` at that point earlier but found that it was not containing `RuntimePackAsset` paths. But I think I made some mistake in my testing.

Let me know if you have more feedback. Thanks.